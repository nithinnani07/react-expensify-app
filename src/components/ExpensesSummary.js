import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) => (
    <div>
        <p>Viewing { props.expenses.length } expenses totalling { numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00') }</p>}
    </div>
);

const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpensesSummary);