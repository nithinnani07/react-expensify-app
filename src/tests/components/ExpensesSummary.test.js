import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render summary text with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});


test('should render summary text with many expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={10} expensesTotal={235001} />);
    expect(wrapper).toMatchSnapshot();
});