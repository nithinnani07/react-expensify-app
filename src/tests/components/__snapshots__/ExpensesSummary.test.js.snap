// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render summary text with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should render summary text with many expenses 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totalling 
    $2,350.01
  </h1>
</div>
`;
